### PR TITLE
ci: Expand test matrices.

### PR DESCRIPTION
**Pull request by stickeeoliver**
Expand test matrix to PHP 8.2. Originally macOS but it's not worth the expense for the time being.

### CODE REVIEW

**Comment by stickeeoliver** [comment]
Looks like we have failing tests... best to get these resolved first before expanding testing matrix.

**Comment by stickeeoliver** [comment]
Shelving this until PHP CS Fixer officially supports 8.2.

**Comment by stickeepaul** [comment]
`https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/6714#issuecomment-1359166470`

> PHP CS Fixer is not ready to run on 8.2.

we'll need to wait a bit more

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -9,8 +9,8 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        os: [ ubuntu-latest, macos-latest ]
-        php: [ 8.1, ]
+        os: [ ubuntu-latest, macos-latest, windows-latest ]
```
Comment (stickeepaul) [inline .github/workflows/tests.yaml]:
```suggestion
        os: [ ubuntu-latest ]
```
let's drop MacOS and Windows

**Review by stickeepaul — CHANGES_REQUESTED** [review]

**Comment by stickeepaul** [comment]
PHP CS fixer now supports PHP 8.2
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/releases/tag/v3.15.0

**Review by stickeepaul — APPROVED** [review]

**Pull request merged**